tests/unit/io/SGIO_unittest.cc

```diff
@@ -0,0 +1,68 @@
+#include <cmath>
+#include <thread>
+#include <chrono>
+#include <shogun/io/SGIO.h>
+
+#include <gtest/gtest.h>
+
+using namespace shogun;
+
+const int millis= 10;
+
+TEST(SGIOTest, progress_correct_bounds)
+{
+	/* Positive values */
```

Comment:
pls split positive and negative into separate tests. tests should be as small as possible